### PR TITLE
Failing operation when panic occurs in the step

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

Changes proposed in this pull request:

- `RunStep` in staged_manager handles panic
- `RunStep` in upgrade_cluster/manager handles panic
- `RunStep` in upgrade_kyma/manager handles panic

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/process/staged_manager.go [hunk 1]
```diff
@@ -190,13 +194,24 @@ func (m *StagedManager) saveFinishedStage(operation internal.Operation, s *stage
...
+
 	begin := time.Now()
 	for {
-		start := time.Now()
+		start = time.Now()
 		logger.Infof("Start step")
-		processedOperation, when, err := step.Run(operation, logger)
-
+		processedOperation, backoff, err = step.Run(operation, logger)
+		//		return s.operationManager.OperationFailed(operation, "call to the provisioner service failed", err, log)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/staged_manager.go:213]: remove the comment

Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/process/staged_manager.go]: Removed

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**